### PR TITLE
add seal fn back to ServerStyleSheet for backward compat

### PR DESCRIPTION
**Pull request by probablyup**

### CODE REVIEW

File: packages/styled-components/src/models/ServerStyleSheet.js [hunk 1]
```diff
@@ -110,4 +101,8 @@ export default class ServerStyleSheet {
 
     return readableStream.pipe(transformer);
   }
+
+  seal() {
+    this.sealed = true;
```
Comment (probablyup) [inline packages/styled-components/src/models/ServerStyleSheet.js]: doesn't really change anything as far as I can see, but just here for API backward compat

**Comment by probablyup** [comment]
@kitten any comments on this one?

File: packages/styled-components/src/models/ServerStyleSheet.js [hunk 2]
```diff
@@ -66,7 +58,7 @@ export default class ServerStyleSheet {
     }
 
     return <style {...props} />;
-  }
+  };
 
   interleaveWithNodeStream(input: any) {
```
Comment (kitten) [inline packages/styled-components/src/models/ServerStyleSheet.js]: I guess we do want to readd setting seal to true. But we can get away with setting it right at the top before the stream and only throwing an error here I think

Comment (probablyup) [reply packages/styled-components/src/models/ServerStyleSheet.js]: @kitten want to push onto my PR? According to the tests sealed is being set correctly in an error scenario so not 100% what you want

**Comment by probablyup** [comment]
@kitten ok I refactored ServerStyleSheet a tad to allow what we need while still enforcing the error scenarios against the public API

**Pull request merged**